### ISSUE TITLE
Modrinth Freezes at `pnpm build` 

### ISSUE THREAD

**Issue by Eveeifyeve** [NixOS/nixpkgs#352586]
### Describe the bug
Basically pnpm goes stuck at builiding vite and according to btop it is sleeping.

My config files are [here](https://github.com/eveeifyeve/dotfiles)
That if you wanted to checkout what is freazing.

### Steps To Reproduce
1. Add `modrinth` to home manager packages.
2. Rebuild your flake config using `sudo nixos-rebuild switch --flake <path> `
3. Open btop and it says `pnpm` is sleeping

### Screenshots
Btop:
![image](https://github.com/user-attachments/assets/6239c1de-4e93-4038-a5a5-8b3927d24098)

[Nh](https://github.com/viperML/nh) Build 
![image](https://github.com/user-attachments/assets/b650ac99-4031-42bb-b353-d41b64cbb2db)

### Additional context
Just to note it's not a nh problem I have tried it without nh and the same thing happens.

### Notify maintainers

@getchoo 

### Metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.58, NixOS, 24.11 (Vicuna), 24.11.20241027.0fcb98a`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/93haqkb7bl8l90vpfqgkygrd5l13gafm-source`

```
---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by gepbird** [comment]
I couldn't reproduce the issue with `NIXPKGS_ALLOW_UNFREE=1 nom build nixpkgs/2d2a9ddbe3f2c00747398f3dc9b05f7f2ebb0f53#modrinth-app --impure` (commit rev taken from your dotfiles flake update), if you try again it should download it from the binary cache rather than build it locally.

I have no idea why pnpm/turbo was just doing nothing.

**Comment by FliegendeWurst** [comment]
Duplicate of #353709, I think. That issue has more details, so I'm closing this one.

**Comment by gador** [comment]
> I couldn't reproduce the issue with `NIXPKGS_ALLOW_UNFREE=1 nom build nixpkgs/2d2a9ddbe3f2c00747398f3dc9b05f7f2ebb0f53#modrinth-app --impure` (commit rev taken from your dotfiles flake update), if you try again it should download it from the binary cache rather than build it locally.
> 
> I have no idea why pnpm/turbo was just doing nothing.

This depends on the build machine, not on the derivation. This is why this fails/builds on some machines, but not on others. As @FliegendeWurst said, the other issue has more info

**Comment by Eveeifyeve** [comment]
> > I couldn't reproduce the issue with `NIXPKGS_ALLOW_UNFREE=1 nom build nixpkgs/2d2a9ddbe3f2c00747398f3dc9b05f7f2ebb0f53#modrinth-app --impure` (commit rev taken from your dotfiles flake update), if you try again it should download it from the binary cache rather than build it locally.
> > I have no idea why pnpm/turbo was just doing nothing.
> 
> This depends on the build machine, not on the derivation. This is why this fails/builds on some machines, but not on others. As @FliegendeWurst said, the other issue has more info

Specs are:

CPU: AMD Ryzen 3 3200g.
Motherboard: B450 MH dash
Disk: Seagate ST2000DM008 2TB 3.5-inch Hard Drive BarraCuda HDD 3.5"

**Comment by gador** [comment]
@Eveeifyeve your build machine specs do not matter. Please read the linked bug report

**Comment by Eveeifyeve** [comment]
This is occuring in kernal 6.6.59 which is two versions after the patch, I think it already reported but I don't know if this is a modrinth package issue or https://github.com/NixOS/nixpkgs/issues/359376

**Comment by FliegendeWurst** [comment]
You need 6.6.60

**Comment by Eveeifyeve** [comment]
Fixed at kernal `6.6.60`.
